src/main/java/address/Browser/BrowserManager.java

```diff
@@ -1,53 +1,177 @@
+                    }
+                }
+            });
+        }
     }
 
     /**
-     * Frees resources allocated to the browser.
+     * Initialise browser to initial state.
      */
-    public void freeBrowserResources(){
-        browser.dispose();
+    public void initBrowser() {
+        for (int i = 0; i < browser.getNumberOfPages(); i++){
+            browser.getPage(i).loadHTML(INITIAL_PAGE_HTML_CODE);
+        }
     }
 
-    public Browser getBrowser(){
-        return browser;
+    /**
+     * Loads the person's GitHub profile page to the browser.
+     * PreCondition: ModelManager getFilteredPersons method returns more than one person.
+     * @param person
+     */
+    public void loadGitHubProfilePage(Person person) {
+
+        List<Person> filteredPersons = modelManager.getFilteredPersons();
+        int indexOfContactsInFilteredListView = filteredPersons.size();
+
+        int indexOfSelectedTabBrowserPage = findIndexOfTabBrowserThatContainsPersonGitHubWebPage(person);
+
+        if (indexOfSelectedTabBrowserPage == PERSON_DOES_NOT_EXISTS_IN_BROWSER) {
+            indexOfSelectedTabBrowserPage = findASuitableBrowserTabAndLoadThePersonGitHubProfilePage(
+                                            person, filteredPersons, indexOfContactsInFilteredListView);
+        }
+        browser.selectTab(indexOfSelectedTabBrowserPage);
+
+        updateOtherBrowserTabs(filteredPersons, indexOfSelectedTabBrowserPage);
+    }
+
+    /**
+     * Returns the index of the tab in the TabBrowser that contains the person's GitHub profile page.
+     * @param person
+     * @return the index of the tab in the TabBrowser that contains the person's GitHub profile page
+     *         or -1 if the tabs of the TabBrowser do not contain the person's GitHub profile page.
+     */
+    private int findIndexOfTabBrowserThatContainsPersonGitHubWebPage(Person person) {
+        int indexOfSelectedTabBrowserPage = PERSON_DOES_NOT_EXISTS_IN_BROWSER;
+        for (TabBrowserPage page : browser.getAllPages()) {
+            if (!page.isAssignedToLoadAPersonGitHubProfilePage()) {
+                continue;
+            }
+            if (page.getPerson().equals(person)) {
+                indexOfSelectedTabBrowserPage = browser.indexOf(page);
+                break;
+            }
+        }
+        return indexOfSelectedTabBrowserPage;
     }
 
-    private void automateClickingAndScrolling() {
-        DOMElement container = browser.getDocument().findElement(By.id("js-pjax-container"));
-        DOMElement link = browser.getDocument().findElement(By.className("octicon octicon-repo"));
-        if (link != null) {
-            container.addEventListener(DOMEventType.OnLoad, e ->
+    private int findASuitableBrowserTabAndLoadThePersonGitHubProfilePage(Person person, List<Person> persons,
+                                                                         int indexOfContactsInFilteredListView) {
+
+        int indexOfSelectedTabBrowserPage = 0;
```

Comment:
I think here to line 114 can be extracted into another method e.g. `getUnassignedTabIndex`. Also try to simplify the logic if possible, since it seems a bit messy.